apps/framework-cli/src/project.rs

```diff
@@ -11,111 +11,146 @@
 //! - `project_file_location` - The location of the project file on disk
 //! ```
 
-use std::collections::HashMap;
 use std::io::Write;
+pub mod project_config_file;
+pub mod typescript_project;
+
+use std::fmt::Debug;
+use std::path::Path;
 use std::path::PathBuf;
 
+use config::{Config, ConfigError, File};
+use log::debug;
+use serde::Serialize;
+
 use crate::cli::local_webserver::LocalWebserverConfig;
 use crate::framework::languages::SupportedLanguages;
 use crate::framework::readme::BASE_README_TEMPLATE;
 use crate::framework::schema::templates::BASE_MODEL_TEMPLATE;
 use crate::infrastructure::console::ConsoleConfig;
 use crate::infrastructure::olap::clickhouse::config::ClickhouseConfig;
 use crate::infrastructure::stream::redpanda::RedpandaConfig;
-
-use crate::utilities::constants::{
-    APP_DIR, APP_DIR_LAYOUT, CLI_PROJECT_INTERNAL_DIR, PROJECT_CONFIG_FILE_TS, SCHEMAS_DIR,
-};
-use config::{Config, ConfigError, File};
-use log::debug;
-use serde::{Deserialize, Serialize};
-use std::path::Path;
-
-// TODO: Move package json file to a typescript module
-#[derive(Serialize, Deserialize, Debug, Clone)]
-#[serde(rename_all = "camelCase")]
-struct PackageJsonFile {
-    pub name: String,
-    pub version: String,
-    pub scripts: HashMap<String, String>,
-    pub dependencies: HashMap<String, String>,
-    pub dev_dependencies: HashMap<String, String>,
+use crate::project::project_config_file::ProjectConfigFile;
+use crate::project::typescript_project::TypescriptProject;
+
+use crate::utilities::constants::PROJECT_CONFIG_FILE;
+use crate::utilities::constants::{APP_DIR, APP_DIR_LAYOUT, CLI_PROJECT_INTERNAL_DIR, SCHEMAS_DIR};
+
+// We have explored using a Generic associated Types as well as
+// Dynaimc Dispatch to handle the different types of projects
+// the approach with enums is the one that is the simplest to put into practice and
+// maintain. With Copilot - it also has the advaantage that the boiler plate is really fast to write
+#[derive(Debug)]
+pub enum Project {
+    Typescript(TypescriptProject),
```

Comment:
The default design I'd go for would be to put only language specific stuff in the enum.

Reply:
Do you have a suggestion on how you would do it?

Reply:
🤔

```
pub struct TypescriptProjectConfig {
    // we don't have language specific config right now
}

pub enum LanguageConfig {
    Typescript(TypescriptProjectConfig)
}

pub struct Project {
    pub name: String,
    pub language_config: LanguageConfig,
    ...
}
```

Reply:
I guess the downside of that is the de-serialization can't be auto-generated.

Reply:
Good suggestion!

I will try to bake that in